include/net/socket.h

```diff
@@ -31,14 +31,24 @@
 extern "C" {
 #endif
 
+struct timeval;
+
 struct zsock_pollfd {
 	int fd;
 	short events;
 	short revents;
 };
 
+/* This implementation uses O(n) operations. Avoid select() in preference
+ * of poll().
+ */
+typedef struct zsock_fd_set {
+	uint32_t bitset[(CONFIG_NET_SOCKETS_POLL_MAX + 31) / 32];
```

Comment:
> Is CONFIG_NET_SOCKETS_POLL_MAX actually related to the bitset size?

In an abstract case, they're all independent, but in reality, they all correlate. So, instead of defining CONFIG_FD_MAXSIZE, I just took something already existing.

Re: specifically CONFIG_POSIX_MAX_FDS, checking that I actually took care that it's defined even if CONFIG_POSIX_API isn't - then you're right, it's better suited, thanks for a hint, fixing.

This all can be of course extended further - poll() and select() shouldn't really live in net/lib/, they should live somewhere close to lib/posix/, but moving it there involves quite a bunch of logistics due to all of Zephyr's cmake usage, POSIX subsys, socket subsys peculiarities.